xattr dependency

When installing conu from the PyPI, the `xattr` package is not installed.

```
/usr/lib/python3.6/site-packages/conu/backend/docker/container.py:35: in <module>
    from conu.apidefs.filesystem import Filesystem
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

    import logging
    import os
    import shutil
>   import xattr
E   ModuleNotFoundError: No module named 'xattr'
```

Is adding `xattr` to [requirements.txt](https://github.com/user-cont/conu/blob/master/requirements.txt) enough?

----

Or make it optional?

--- a/conu/apidefs/filesystem.py
+++ b/conu/apidefs/filesystem.py
@@ -17,8 +17,13 @@
 import logging
 import os
 import shutil
-import xattr
 from tempfile import mkdtemp
+
+try:
+    import xattr
+    HAS_XATTR = True
+except ImportError:
+    HAS_XATTR = False
 
 from conu.exceptions import ConuException
 
@@ -154,11 +159,16 @@ class Filesystem(object):
 
     def get_selinux_context(self, file_path):
         """
-        return a permissions for 'file_path'
+        Get SELinux file context of the selected file.
 
         :param file_path: str, path to the file
-        :return: str
+        :return: str, name of the SELinux file context
         """
         # what if SELinux is not enabled?
         p = self.p(file_path)
+        if not HAS_XATTR:
+            raise RuntimeError("'xattr' python module is not available, hence we cannot "
+                               "determine the SELinux context for this file. "
+                               "In Fedora this module is available as python3-pyxattr -- "
+                               "other distributions may follow similar naming scheme.")
         return xattr.get(p, "security.selinux")

--- a/requirements.in
+++ b/requirements.in
@@ -3,6 +3,5 @@ requests
 six
 docker
 kubernetes >= 6.0.0
-pyxattr
 # conu/fixtures/
 pytest

--- a/requirements.sh
+++ b/requirements.sh
@@ -18,11 +18,8 @@ if [ "${NAME}" == "Fedora" ]; then
         source-to-image \
         origin-clients \
         python3-pip python2-pip \
-        python3-pyxattr \
+        python3-pyxattr python2-pyxattr \
         python3-pytest python2-pytest \
         gcc python3-devel python2-devel \
         make
-
-    # It has different naming in Fedora 27 and Fedora 28
-    dnf install -y python2-pyxattr || dnf install -y pyxattr
 fi

--- a/tests/integration/test_filesystem.py
+++ b/tests/integration/test_filesystem.py
@@ -160,3 +160,9 @@ def test_image_dir_is_present(docker_image):
         assert not fs.directory_is_present("/etc/voldemort")
         with pytest.raises(IOError):
             fs.directory_is_present("/etc/passwd")
+
+
+@pytest.mark.selinux
+def test_selinux_context(docker_image):
+    with docker_image.mount() as fs:
+        assert fs.get_selinux_context("/etc/shadow")

